### PR TITLE
Bump: sentry-android to v4.3.0

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Bump: sentry-android to v4.3.0

## :bulb: Motivation and Context
Kotlin nullability and sentry-native improvements

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: flutter/android/proguard-rules.pro [hunk 1]
```diff
@@ -1 +1,5 @@
 -keep class io.sentry.flutter.** { *; }
+
+# To ensure that stack traces is unambiguous
+# https://developer.android.com/studio/build/shrink-code#decode-stack-trace
```
Comment (philipphofmann) [inline flutter/android/proguard-rules.pro:4]: Thanks for the important comment 💯

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 2]
```diff
@@ -168,12 +168,12 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
   private val nativeSdk = "sentry.native"
 
   private fun setEventOriginTag(event: SentryEvent) {
-    val sdk = event.sdk
-    if (isValidSdk(sdk)) {
-      when (sdk!!.name) {
+    event.sdk?.let {
+      when (it.name) {
```
Comment (philipphofmann) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt:172]: Way more precise 👍🏼

**Review by philipphofmann — APPROVED** [review]
ktlint is complaining a bit. Apart from that LGTM.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Unreleased
 
+* Bump: sentry-android to v4.3.0
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Bump: sentry-android to v4.3.0 (#343)
```

**Review by bruno-garcia — COMMENTED** [review]
LGTM

**Pull request merged**